### PR TITLE
[superlu] Add new port

### PR DESCRIPTION
**Pull request by otakuto**
- What does your PR fix?
Add a Supernodal sparse direct solver library.

- Which triplets are supported/not supported? Have you updated the CI baseline?
No.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes.

### CODE REVIEW

**Comment by NancyLi1013** [comment]
Hi @otakuto thanks for this PR.
I noticed that it failed on windows platform. Some errors like this:
```
SRC\CMakeFiles\superlu.dir\zdiagonal.c.obj C:\vsts\_work\4\s\installed\x86-windows\debug\lib\openblas.lib m.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib 
LINK : fatal error LNK1104: cannot open file 'm.lib'
ninja: build stopped: subcommand failed.
```
Could you please take a look and try to fix the regressions?

**Comment by NancyLi1013** [comment]
Since this port doesn't support `arm, arm64, uwp`.
Could you also please update 
```
superlu:x64-uwp=fail
superlu:arm64-windows=fail
superlu:arm-uwp=fail
```
to [ci.baseline.txt](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?

**Comment by otakuto** [comment]
Thanks.
I will do that.

**Review by NancyLi1013 — APPROVED** [review]

**Review by PhoebeHui — APPROVED** [review]
Thanks for the PR!

File: ports/superlu/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      remove-make.inc.patch
+)
+
+string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" SUPERLU_SHARED)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+    OPTIONS
+    -DBUILD_SHARED_LIBS=${SUPERLU_SHARED}
```
Comment (ras0219-msft) [inline ports/superlu/portfile.cmake]: This is passed automatically by `vcpkg_configure_cmake()` and does not need to be explicitly added.

File: ports/superlu/fix-libm.patch [hunk 2]
```diff
@@ -0,0 +1,17 @@
+diff --git a/SRC/CMakeLists.txt b/SRC/CMakeLists.txt
+index e627f06..9aa8949 100644
+--- a/SRC/CMakeLists.txt
++++ b/SRC/CMakeLists.txt
+@@ -232,7 +232,11 @@ if(enable_complex16)
+ endif()
+ 
+ add_library(superlu ${sources} ${HEADERS})
+-target_link_libraries(superlu PUBLIC ${BLAS_LIB} m)
++if(${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
```
Comment (ras0219-msft) [inline ports/superlu/fix-libm.patch]: This potentially double-expands the variable -- it should probably be
```cmake
if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
```

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

**Comment by otakuto** [comment]
@ras0219-msft 
Thanks for your review.
Comments fixed.

**Comment by otakuto** [comment]
ping

**Review by NancyLi1013 — APPROVED** [review]

**Review by PhoebeHui — APPROVED** [review]
Thanks for the updates!

**Comment by strega-nil** [comment]
Thanks @otakuto :D Looks awesome

**Pull request merged**